man/fuse.1
.\" -*- nroff -*-
.\"
.\" fuse.1: Fuse man page
.\" Copyright (c) 2001-2013 Russell Marks, Philip Kendall, Darren Salt,
.\"                         Fredrick Meunier, Stuart Brady
.\"
.\" This program is free software; you can redistribute it and/or modify
.\" it under the terms of the GNU General Public License as published by
.\" the Free Software Foundation; either version 2 of the License, or
.\" (at your option) any later version.
.\"
.\" This program is distributed in the hope that it will be useful,
.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License along
.\" with this program; if not, write to the Free Software Foundation, Inc.,
.\" 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
.\"
.\" Author contact information:
.\"
.\" E-mail: philip-fuse@shadowmagic.org.uk
.\"
.\"
.TH fuse 1 "24th May, 2013" "Version 1.1.1" "Emulators"
.\"
.\"------------------------------------------------------------------
.\"
.SH NAME
fuse \(em Sinclair ZX Spectrum emulator
.\"
.\"------------------------------------------------------------------
.\"
.SH SYNOPSIS
.\" the trend for long-option-using programs is to give a largely
.\" generic synopsis, so...
.B fuse
.RI [ options ]
.P
.\"
.\"------------------------------------------------------------------
.\"
.SH DESCRIPTION
Fuse is a Sinclair ZX\ Spectrum emulator. It supports several models
(including the 128), with quite faithful emulation of the display and
sound.
.PP
The emulator can load any of the formats supported by
.IR libspectrum (3)
\(em this includes Z80, SNA and SZX snapshots, and PZX, TAP and TZX
virtual-tape files. Saving to SZX, Z80 and SNA snapshots and TZX and
TAP tape files is supported. The SLT extension to the Z80 format is
partly supported (enough for multi-load games); however, loading of
the old DAT-file variant is not. DSK, UDI, FDI, TR0, SDF, MGT, IMG,
SAD, TRD, SCL and OPD disk images are supported when a disk interface
is being emulated, including the integrated disk drives on +3,
Pentagon or Scorpion machines as well as the +D, DISCiPLE,
Opus Discovery and Beta\ 128 interfaces. DCK cartridge images are
supported when emulating a Timex\ 2068 variant. Interface\ 2 ROM
cartridges are also supported.
.PP
Finally, there is also support for reading and writing the RZX input recording
format.
.PP
See the
.B "COMPRESSED FILES"
section for details on reading files compressed with
.IR bzip2 "(3)"
or
.IR gzip "(3)."
.\"
.\"------------------------------------------------------------------
.\"
.SH OPTIONS
.\" dual short/long option listings here reflect the GNU approach,
.\" as used in info files. It does mean using RS/RE, though,
.\" so this is used for all options (for consistency).
.\"
.\" the options list is in alphabetical order by long option name (or
.\" short option name if none).
.\"
.B \-\-accelerate\-loader
.RS
Specify whether Fuse should attempt to accelerate tape loaders by \(lqshort
circuiting\(rq the loading loop. This will in general speed up loading, but
may cause some loaders to fail. (Enabled by default, but you can use
.RB ` \-\-no\-accelerate\-loader '
to disable). The same as the General Options dialog's
.I "Accelerate loaders"
option.
.RE
.PP
.B \-\-aspect\-hint
.RS
Specify whether the GTK+ and Xlib user interfaces should `hint' to the
window manager about the preferred aspect ratio for the graphics
window, thus preventing resizing to non-square sizes which lead to
Fuse not displaying correctly. This option has been observed to cause
problems with some window managers when using the GTK+ UI which can
prevent the window from being resized or moved at all. (Enabled by
default, but you can use
.RB ` \-\-no\-aspect\-hint '
to disable). See also the
.RB ` \-\-strict\-aspect\-hint '
option.
.RE
.PP
.B \-\-autosave\-settings
.RS
Specify whether Fuse's current settings should be automatically saved
on exit. The same as the General Options dialog's
.I "Auto-save settings"
option.
.RE
.PP
.B \-\-auto\-load
.RS
Specify whether tape and disk files should be automatically loaded
when they are opened using the
.I "File, Open..."
menu option. (Enabled by default, but you can use
.RB ` \-\-no\-auto\-load '
to disable). Same as the General Options dialog's
.I "Auto-load media"
option.
.RE
.PP
.B \-\-beta128
.RS
Emulate a Beta\ 128 interface. Same as the Disk Peripherals Options dialog's
.I "Beta\ 128 interface"
option.
.RE
.PP
.B \-\-beta128\-48boot
.RS
When a Beta\ 128 interface is used in 48K or TC2048 emulation the
option additionally controls whether the machine boots directly into the TR-DOS
system. Same as the Disk Peripherals Options dialog's
.I "Beta\ 128 auto-boot in 48K machines"
option.
.RE
.PP
.B \-\-betadisk
.I file
.RS
Insert the specified file into the emulated Beta disk interface's drive\ A:
and select Pentagon mode on startup.
.RE
.PP
.B \-\-bw\-tv
.RS
Specify whether the display should simulate a colour or black and
white television. This option is effective under the GTK+, Win32, Xlib
and SDL user interfaces: the others will always simulate a colour TV.
The same as the General Options dialog's
.I "Black and white TV"
option.
.RE
.PP
.B \-\-competition\-code
.I code
.RS
Specify the code to be written to competition mode RZX files. The same
as the RZX Options dialog's
.I "Competition code"
option.
.RE
.PP
.B \-\-competition\-mode
.RS
Specify whether input recordings should be made in `competition mode'.
The same as the RZX Options dialog's
.I "Competition mode"
option.
.RE
.PP
.B \-\-compress\-rzx
.RS
Specify whether RZX files should be written out compressed. (Enabled
by default, but you can use
.RB ` \-\-no\-compress\-rzx '
to disable). Same as the RZX Options dialog's
.I "Compress RZX data"
option.
.RE
.PP
.B \-\-confirm\-actions
.RS
Specify whether `dangerous' actions (those which could cause data
loss, for example resetting the Spectrum) require confirmation before
occurring. (Enabled by default, but you can use
.RB ` \-\-no\-confirm\-actions '
to disable). This option is the same as the General Options dialog's
.I "Confirm actions"
option.
.RE
.PP
.B \-\-debugger\-command
.I string
.RS
Specify a debugger command to be run before emulator startup. This can
be used to set breakpoints or the like. Currently, this is the only
method to input multi-line debugger commands. (See the
.B "MONITOR/DEBUGGER"
section for more information).
.RE
.PP
.B \-\-detect\-loader
.RS
Specify whether Fuse should attempt to detect when the tape is being
accessed and start and stop the virtual tape playing
automatically. (Enabled by default, but you can use
.RB ` \-\-no\-detect\-loader '
to disable). Same as the General Options dialog's
.I "Detect loaders"
option.
.RE
.PP
.B \-\-disciple
.RS
Emulate a DISCiPLE interface. Same as the Disk Peripherals Options dialog's
.I "DISCiPLE interface"
option.
.RE
.PP
.B \-\-discipledisk
.I file
.RS
Insert the specified file into the emulated DISCiPLE's drive\ 1.
.RE
.PP
.B \-\-disk\-ask\-merge
.RS
Prompt the user to confirm whether Fuse should try to merge the `B' side of a
disk image from a separate file when opening a new single-sided disk image.
.RE
.PP
.B \-\-disk\-try\-merge
.I mode
.RS
Select whether Fuse should try to merge a separate file for the `B' side of a
disk image separate file when opening a new disk image. Most double sided disk
images are dumped as two single sided disk images e.g.
`Golden Axe \- Side\ A.dsk' and `Golden Axe \- Side\ B.dsk'. So, if we want to
play Golden Axe, first we have to insert the first disk image and when the game
asks to insert side\ B, we have to find and open the second disk image, instead
of just `flip'-ing the disk inside the drive. If enabled, Fuse will try to open
the second image too and create a double sided disk image (merging the two one
sided disk images) and insert this merged virtual disk into the disk drive. The
function detects whether the file is one side of a double-sided image if the
filename matches a pattern like [Ss]ide[ _][abAB12][ _.] in the file name of a
disk that is being opened. If found, Fuse will try to open the other side of
the disk too substituting the appropriate characters in the filename e.g.
1\(->2, a\(->b, A\(->B. If successful then it will merge the two images and
now we have a double sided disk in drive. This means that if we open
`Golden Axe \- Side\ A.dsk', then Fuse will try to open `Golden Axe \- Side
B.dsk' too. Now, we can just `flip' the disk if Golden Axe asks for `Side\ B'.
The available options are
.IR Never ,
.I "With single-sided drives"
and
.IR Always .
.RE
.PP
.B \-\-divide
.RS
Emulate the DivIDE interface. The same as the Disk Peripherals Options
dialog's
.I "DivIDE interface"
option.
.RE
.PP
.B \-\-divide\-masterfile
.I file
.br
.B \-\-divide\-slavefile
.I file
.RS
Specify an IDE image to be loaded into the DivIDE's emulated master
and slave drives respectively.
.RE
.PP
.B \-\-divide\-write\-protect
.RS
Specify that the emulated DivIDE's write protect jumper should be
considered set. The same as the Disk Peripherals Options dialog's
.I "DivIDE write protect"
option.
.RE
.PP
.B \-\-dock
.I file
.RS
Insert the specified file into the emulated Timex\ 2068 variant dock;
also select the TC2068 on startup if available.
.RE
.PP
.B \-\-doublescan\-mode
.RS
Specify that the framebuffer UI should attempt to use a double scan mode
(where each line is displayed twice).
.RE
.PP
.B \-\-drive\-plus3a\-type
.I type
.br
.B \-\-drive\-plus3b\-type
.I type
.br
.B \-\-drive\-beta128a\-type
.I type
.br
.B \-\-drive\-beta128b\-type
.I type
.br
.B \-\-drive\-beta128c\-type
.I type
.br
.B \-\-drive\-beta128d\-type
.I type
.br
.B \-\-drive\-plusd1\-type
.I type
.br
.B \-\-drive\-plusd2\-type
.I type
.br
.B \-\-drive\-disciple1\-type
.I type
.br
.B \-\-drive\-disciple2\-type
.I type
.br
.B \-\-drive\-opus1\-type
.I type
.br
.B \-\-drive\-opus2\-type
.I type
.RS
Specify a disk drive type to emulate with the associated interface. 
See the Disk Options dialog for more information.
.RE
.PP
.B \-\-drive\-40\-max\-track
.I count
.br
.B \-\-drive\-80\-max\-track
.I count
.RS
Specify the maximum number of tracks for 40 and 80 track physical drives
respectively.
.RE
.PP
.B \-\-embed\-snapshot
.RS
Specify whether a snapshot should be embedded in an RZX file when
recording is started from an existing snapshot. (Enabled by default,
but you can use
.RB ` \-\-no\-embed\-snapshot '
to disable). Same as the RZX Options dialog's
.I "Always embed snapshot"

ChangeLog
2013-05-xx  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * libspectrum 1.1.1 released.

        * Decreased libtool version to allow binaries built against
          libspectrum 1.0.0 to run with libspectrum 1.1.1.

2013-05-18  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * libspectrum 1.1.0 released.

        * Support for Spectranet in snapshots (Philip Kendall).

        * Support for Cheetah SpecDrum in snapshots (Fredrick Meunier).

        * Support reading PZX files (Fredrick Meunier).

        * Support for the TZX set signal level block (Fredrick Meunier).

        * Various minor bug fixes/improvements:
          * Add support for detecting POK files (Sergio Baldoví).
          * Advertise support for bzip2 compressed and wav files (Sergio
            Baldoví).
          * Drop support for GLib version 1.x (Alberto Garcia).
          * Add support for audiofile 0.3.x (Alberto Garcia).
          * Load all snapshot RAM pages for machines with more than 128k 
            (Fredrick Meunier; thanks, ketmar).
          * Support more SZX 1.5 features (Fredrick Meunier).
          * Modernise autoconf support (Sergio Baldoví).

2010-12-16  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * libspectrum 1.0.0 released.

        * New end of tape flag (Fredrick Meunier).

        * AF and AF' were swapped when reading and writing SZX files
          (Fredrick Meunier; thanks, velesoft).

        * Support for Fuller Box and Melodik (and similar AY interfaces) in
          snapshots (Fredrick Meunier).

        * Support for Opus Discovery in snapshots (Fredrick Meunier, Gergely
          Szasz).

        * Support for the NTSC 48K Spectrum (Philip Kendall).

        * Support for stereo sound files (Fredrick Meunier; thanks, Gilberto
          Almeida).

        * Add proper support for +2A/+3, Pentagon and Scorpion in .z80 files
          (Philip Kendall; thanks, Nick Corey)

        * Various fixes for compiling with Visual Studio 6 (Simon Owen).

        * Various minor bug fixes/improvements:
          * Looser signature checks for .dsk files (Gergely Szasz; thanks,
            zx81 and Simon Owen).
          * --without-(bzip2|zlib) options to configure for Gentoo and similar
            distributions (José Manuel Ferrer Ortiz).
          * Handle querying the block length for turbo and pure data blocks
            with 0 length (Fredrick Meunier).
          * Fix loading of Warajevo TAPs with raw samples (Fredrick Meunier).
          * Fail sensibly if file decompression fails (Fredrick Meunier).
          * Remove various unused variables, initialisations etc (Fredrick
            Meunier).

2008-11-30  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * libspectrum 0.5.0 released.

	* Memory management change: memory allocation functions now abort
	  on failure. Leads to simpler code both internally and for
	  users of the library (Philip Kendall).

	* Support for tape acceleration (Philip Kendall)

	* IDE improvements to allow DivIDE to support FATware (Philip
	  Kendall; thanks, http://www.dusky.sk/zxs/)

	* New API to deal with flagging automatic snapshots in RZX files
	  (Philip Kendall).

	* New function to get length (time) of a tape block (Philip Kendall,
	  Fredrick Meunier).

	* 128K SNA files are now treated as being for the Pentagon 128
	  rather than the Spectrum 128 (Fredrick Meunier)

	* SNA files now written out correctly (Philip Kendall).

	* Handle MDR write protection flag correctly (Philip Kendall)

	* Allow MDR images with bad checksums to load (the error will
	  be handled by the If1 ROM) (Gergely Szasz)

        * Support saving custom ROMs in SZX snapshots (Fredrick Meunier,
	  Stuart Brady)

	* Support for DivIDE, Kempston mouse and the simple IDE interface
	  in SZX snapshots (Fredrick Meunier)

        * Allow building without libgcrypt (Stuart Brady)

	* Amiga and MorphOS compilation fixes (Chris Young, Q-Master)

	* Don't produce empty libraries to fix build process on
	  OpenSolaris (Fredrick Meunier)

	* Remove deprecated functions (Fredrick Meunier).

	* Tweak timings for TS2068 (Fredrick Meunier; thanks Philip Kendall).

        * NTSC Spectrum support (Philip Kendall; thanks, Claudio Bernet).

2008-01-05  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

	* libspectrum 0.4.0 released.

	* Add support for loading from WAV files (Fredrick Meunier).

	* Add support for CSW tape files (Fredrick Meunier).

	* Tweak timings for TC2048 and TC2068 (Fredrick Meunier; thanks Philip
	  Kendall and Mark Woodmass).

	* Add support for the +D in z80 and szx snapshots (Fredrick Meunier).

	* Add Pentagon 512/1024 support (Q-Master).

	* Allow Amiga version to decompress files with xfdmaster.library
	  (Chris Young).

	* Allow MorphOS version to use xfdmaster.library (Q-Master).

	* Various other bugfixes.

2007-05-11  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

	* libspectrum 0.3.0.1 released (bug fix release).

	* Avoid crash when reading an invalid TZX file (thanks, crabfists).
	  (Philip Kendall).

	* Export the TZX Generalized Data Block functions when building a
	  Win32 DLL (Stuart Brady).

	* Distribute the hacking/ directory (thanks, Stuart Brady) (Philip
	  Kendall).


website/fuse.php
<?php

/* fuse.php: the main Fuse homepage
   Copyright (c) 1999-2008 Darren Salt, Philip Kendall

   $Id$

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License along
   with this program; if not, write to the Free Software Foundation, Inc.,
   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

   Author contact information:

   E-mail: philip-fuse@shadowmagic.org.uk

*/

include ".inc/fuse.inc";

fuse_title ("Fuse - the Free Unix Spectrum Emulator", true);
fuse_menu_heading ("Fuse");

#<!-- ======================================================= -->

$img = mk_image('screens/f-awm.png', 330, 292);
fuse_section (NULL, "What is it?", <<<END_SECTION
  $img
  <p>Fuse (the Free Unix Spectrum Emulator) was originally, and somewhat unsurprisingly, a ZX Spectrum emulator for Unix. However, it has now also been ported to Mac OS X, which may or may not count as a Unix variant depending on your advocacy position. It has also been ported to Windows, AmigaOS and MorphOS, which are definitely not Unix variants.</p>

END_SECTION
);

#<!-- ======================================================= -->

$img = mk_image('screens/f-large.png', 650, 532, 4);
fuse_section ("Features", "What features does it have?", <<<END_SECTION
  $img
  <ul>
   <li>Accurate 16K, 48K, 128K, +2, +2A and +3 emulation.</li>
   <li>Working +3e, SE, TC2048, TC2068, TS2068, Pentagon 128, Pentagon "512" (Pentagon 128 modified for extra memory), Pentagon 1024 and Scorpion ZS 256 emulation.</li>
   <li>Runs at true Speccy speed on any computer you're likely to try it on.</li>
   <li>Support for loading from .tzx files, including accelerated loading.</li>
   <li>Sound (on Windows and Mac OS X, and on systems supporting ALSA, the Open Sound System, SDL or OpenBSD/Solaris's <tt>/dev/audio</tt>).</li>
   <li>Kempston joystick emulation.</li>
   <li>Emulation of the various printers you could attach to the Spectrum.</li>
   <li>Support for the RZX input recording file format, including 'competition mode'.</li>
   <li>Emulation of the DivIDE, Interface I, +D, Beta 128, Kempston mouse, Spectrum +3e, ZXATASP and ZXCF interfaces.</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section ("What's lacking?", "What is it lacking?", <<<END_SECTION
  <ul>
   <li>Quite a lot! However, it's a lot better than it used to be...</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

$img = mk_image('screens/f-rzx.png', 330, 266);
fuse_section ("Download", "Downloads", <<<END_SECTION
  <p>Fuse is licensed under the <a href="COPYING">GNU General Public License</a>, version 2 or later. Please read this before downloading Fuse if you're not already familiar with it.</p>

  <h3>Unix</h3>
  <p>Packages are available for some Unix distributions; in general, any problems which are specific to the packages should be sent to the package maintainer.</p>
  <ul>
   <li>Unofficial packages of 0.10.0.2 for <a href="http://www.debian.org/">Debian</a> (or <a href="http://www.ubuntu.com/">Ubuntu</a>) by Alberto Garcia are available from <a href="http://people.igalia.com/berto/">his personal webpage</a> (i386 binaries and source).</li>
   <li><a href="http://fedoraproject.org/index.html">Fedora</a> has <a href="https://admin.fedoraproject.org/pkgdb/packages/name/fuse-emulator">packages</a> of 0.10.0.1 available.</li>
   <li><a href="http://www.freebsd.org/">FreeBSD</a> has a port of 0.10.0.1 available as <a href="http://www.FreeBSD.org/cgi/cvsweb.cgi/ports/emulators/fuse/">emulators/fuse</a>.</li>
   <li>Nokia's <a href="http://maemo.org/">Maemo</a> platform has a port of <a href="http://maemo.org/downloads/product/OS2008/fuse-emulator/">0.10.0.2</a> available by Alberto Garcia.</li>
   <li><a href="http://www.netbsd.org/">NetBSD</a> users can get version 0.10.0.2 from <a href="http://www.pkgsrc.org/">pkgsrc</a> (the NetBSD Packages Collection) as <a href="http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/emulators/fuse/">emulators/fuse</a> and <a href="http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/emulators/fuse-utils/">emulators/fuse-utils</a>.</li>
   <li><a href="http://www.openbsd.org/">OpenBSD</a> users have version 0.10.0.1 available from ports as <a href="http://www.openbsd.org/cgi-bin/cvsweb/ports/emulators/fuse/">ports/emulators/fuse</a>, and <a href="http://www.openbsd.org/cgi-bin/cvsweb/ports/emulators/fuse-utils/">ports/emulators/fuse-utils/</a>, with thanks to Alexander Yurchenko.</li>
  </ul>
  <p>Packages of older versions of Fuse are also available for some other distributions:</p>
  <ul>
   <li><a href="http://www.gentoo.org/">Gentoo</a> users have an <a href="http://packages.gentoo.org/package/app-emulation/fuse?full_cat">ebuild</a> of 0.9.0 available.</li>
   <li><a href="http://www.mandriva.com/">Mandriva</a> packages of 0.9.0 are available from the <a href="http://plf.zarb.org/">PLF</a>, or Miguel Barrio Orsikowsky has some <a href="http://www.speccy.org/sinclairmania/arch/emu/mandrake/">older versions</a> available.</li>
   <li>Zoran Jekic has produced <a href="http://www.suse.com">SuSE</a> 11.3 packages of <a href="fuse-0.9.0-1.zj.i586.rpm">0.9.0</a> and the <a href="fuse-utils-0.9.0-1.zj.i586.rpm">utilities</a>.</li>
   <li>The <a href="http://en.docs.pld-linux.org/">Polish Linux Distribution</a> has packages of Fuse 0.6.2.1 and the utilities.</li>
   <li><a href="mailto:spec(at)webtech(dot)pl">Marek Januszewski</a> has produced a <a href="http://www.slackware.com/">Slackware</a> 9.1 packages of <a href="fuse-0.6.1.1-i386-1spec.tgz">0.6.1.1</a> (<a href="fuse-0.6.1.1-i386-1spec.tgz.md5">md5 sum</a>) and <a href="fuse-utils-0.6.1-i386-1spec.tgz">the utilities</a> (<a href="fuse-utils-0.6.1-i386-1spec.tgz.md5">md5 sum</a>).</li>
  </ul>

  <h3>Mac OS X</h3>
  <p>A native port to OS X by <a href="mailto:fredm@spamcop.net">Fredrick Meunier</a> is available on its own SourceForge project <a href="http://fuse-for-macosx.sourceforge.net//">here</a>, as well as a Spotlight importer for Mac OS X 10.4 Tiger users. Alternatively, the original version of Fuse will compile on OS X 10.3 (Panther) or later.</p>

  <h3>Windows</h3>
  <p>A port to Windows of 0.10.0.2 by Marek Januszewski and Stuart Brady is available <a href="http://prdownloads.sourceforge.net/fuse-emulator/fuse-0.10.0.2-setup.exe?download">here</a> (<a href="http://prdownloads.sourceforge.net/fuse-emulator/fuse-0.10.0.2-setup.exe.sha1?download">SHA1 hash</a>). This is also mirrored at <a href="http://www.worldofspectrum.org/">World of Spectrum</a>: <a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/pc/windows/fuse-0.10.0.2-setup.exe">Fuse</a> (<a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/pc/windows/fuse-0.10.0.2-setup.exe.sha1">SHA1 hash</a>).</p>

  <h3>AmigaOS 4</h3>
  <p>Chris Young has ported Fuse 0.10.0 to AmigaOS 4, with binaries available from Aminet as <a href="http://aminet.net/package/misc/emu/fuse.lha">misc/emu/fuse.lha</a>.</p>

  <h3>MorphOS</h3>
  <p>Q-Master has ported Fuse 0.10.0.1 to MorphOS, with <a href="http://www.amirus.org.ru/files/fuse_0.10.0.1_morphos.lha">binaries available from AmiRUS</a>.</p>

  <h3>PSP</h3>
  <p>Akop Karapetyan has ported Fuse to the PSP. Binaries and source, based on Fuse 0.10.0.1, are available from the <a href="http://psp.akop.org/fuse">Fuse PSP</a> page.</p>

  <h3>Wii</h3>
  <p>A Wii port, based on work by Bj&ouml;rn Giesler, is available from <a href="http://wiibrew.org/wiki/Fuse">WiiBrew</a>. This is based on what is essentially 0.10.0.2.</p>

  <h3>Gizmondo</h3>
  <p>A port of 0.9.0 to the Gizmondo tablet is available. The source is available via <a href="http://opensvn.csie.org/GizPorts/trunk/sdlport/fuzegiz/">csie.org</a>.</p>

  <h3>GP2X</h3>
  <p>Ben O'Steen has made a GP2X port, based on Fuse 0.6. Binaries and source are available from <a href="http://www.zen71790.zen.co.uk/#fuse">his homepage</a>.</p>

  <h3>XBox</h3>
  <p>Crabfists's has made an Xbox port, based on Fuse 0.6. Binaries and source are available from the <a href="http://sourceforge.net/projects/fusex/">FuseX project</a> at SourceForge.</p>

  <h3>PocketPC</h3>
  <p>Anders Holmberg's ported Fuse 0.4 to the PocketPC as <a href="http://pocketclive.emuunlim.com/">PocketClive</a>.</p>

  <h3>Smartphone</h3>
  <p>Keith Orbell's then ported PocketClive to the Smartphone as <a href="http://www.aooa27.dsl.pipex.com/FuseSP.htm">FuseSP</a>.</p>

END_SECTION
);

#<!-- ======================================================= -->

$img = mk_image('screens/f-nongtk.png', 330, 266);
fuse_section ("Source", "Source", <<<END_SECTION
  <h3>Installing Fuse</h3>
  $img
  <ul>
   <li>First, check the requirements below and ensure all the libraries you want/need are installed.</li>
   <li>Secondly, install <a href="libspectrum.php">libspectrum</a>.</li>
   <li>Get the <a href="http://prdownloads.sourceforge.net/fuse-emulator/fuse-0.10.0.2.tar.gz?download">source code</a> (<a href="http://prdownloads.sourceforge.net/fuse-emulator/fuse-0.10.0.2.tar.gz.sig?download">PGP signature</a>).</li>
   <li>The utilities which were previously packaged with Fuse are now available in their <a href="http://prdownloads.sourceforge.net/fuse-emulator/fuse-utils-0.10.0.1.tar.gz?download">own package</a> (<a href="http://prdownloads.sourceforge.net/fuse-emulator/fuse-utils-0.10.0.1.tar.gz.sig?download">PGP signature</a>). Note that
you'll still need <a href="libspectrum.php">libspectrum</a> installed to run these.</li>
   <li>The above are also mirrored at <a href="http://www.worldofspectrum.org/">World of Spectrum</a>: <a
href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/fuse-0.10.0.2.tar.gz">Fuse source</a> (<a
href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/fuse-0.10.0.2.tar.gz.sig">signature</a>), <a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/fuse-utils-0.10.0.1.tar.gz">utils source</a> (<a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/fuse-utils-0.10.0.1.tar.gz.sig">signature</a>).</li>
   <li>The source code releases above are signed with the <a href="http://wwwkeys.pgp.net:11371/pks/lookup?op=get&amp;search=0xD0767AB9">Fuse Release Key, ID <tt>D0767AB9</tt></a>, which has fingerprint <tt>064E 0BA9 688F 2699 3267 B1E5 1043 EEEB D076 7AB9</tt>. This is different from the key used to sign the 0.6.0(.1) releases as I forgot the passphrase for that key <tt>:-(</tt>.</li>
  </ul>

  <h3>Requirements</h3>
  $img
  <dl>
   <dt>Required:</dt>
   <dd>
    <ul>
     <li>X, <a href="http://www.libsdl.org/">SDL</a>, <a href="http://www.svgalib.org/">svgalib</a> or framebuffer support. If you have <a href="http://www.gtk.org/">GTK+</a> installed, you'll get a (much) nicer user interface under X.</li>
     <li><a href="libspectrum.php">libspectrum</a>: the Spectrum emulator file format and information library.</li>
    </ul>
   </dd>
   <dt>Optional:</dt>
   <dd>
    <ul>
     <li><a href="http://www.gnu.org/directory/security/libgcrypt.html">libgcrypt</a>: the ability to digitally sign RZX files (note that Fuse requires version 1.1.42 or later).</li>
     <li><a href="http://www.libpng.org/pub/png/libpng.html">libpng</a>: the ability to save screenshots.</li>
     <li><a href="http://xmlsoft.org/">libxml2</a>: the ability to load and save Fuse's current configuration.</li>
     <li><a href="http://www.battlefieldlinux.com/wolfpack/libjsw/">libjsw</a>: allow joystick input to be used (not required for joystick emulation).</li>
     <li><a href="http://www.gzip.org/zlib/">zlib</a>: support for compressed RZX files.</li>
     <li><a href="http://www.bzip.org/">libbzip2</a>: support for certain compressed files.</li>
     <li><a href="http://www.68k.org/~michael/audiofile/">libaudiofile</a>: support for loading from .wav files.</li>
     <li><a href="http://www.mega-nerd.com/SRC/">libsamplerate</a>: higher quality sound.</li>
     <li>Versions previous to 0.10.0 used John Elliott's <a href="http://www.seasip.demon.co.uk/Unix/LibDsk/">lib765 and libdsk</a> for the +3 support. 0.10.0 includes this support natively, so these libraries are no longer necessary (or used).</li>
    </ul>
   </dd>
  </dl>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section (NULL, "What's new?", <<<END_SECTION
  <h3>0.10.0.2</h3>
  <ul>
    <li>Make loading acceleration work with all compilers</li>
    <li>A number of minor bug fixes</li>
  </ul>

  <h3>0.10.0.1</h3>
  <ul>
    <li>Stop corruption when overwriting an existing file</li>
    <li>Ensure all header files are distributed</li>
  </ul>

  <h3>0.10.0</h3>
  <ul>
    <li>Improved upd765 FDC emulation</li>
    <li>Loading acceleration</li>
    <li>Automatic saves while RZX recording</li>
    <li>Improved Win32 UI</li>
    <li>Improved widget UI</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section (NULL, "Development", <<<END_SECTION
  <p>If you're just want news of new versions and the like, the (low volume) <a href="http://lists.sourceforge.net/lists/listinfo/fuse-emulator-announce">fuse-emulator-announce</a> list is available. If you're interested in the development of Fuse, this is coordinated via the <a href="http://lists.sourceforge.net/lists/listinfo/fuse-emulator-devel">fuse-emulator-devel</a> list and the <a href="http://sourceforge.net/projects/fuse-emulator/">project page</a> on SourceForge.</p>

  <p>The latest version of Fuse is always available by checking out the 'trunk/fuse' directory from the <a href="http://sourceforge.net/svn/?group_id=91293">Subversion repository</a> on SourceForge. Note that this isn't guaranteed to compile, let alone work properly. Also, don't expect any support for this version! (You'll also need libspectrum from Subversion; this is in the 'trunk/libspectrum' directory). Similarly, the utilities are available in the 'trunk/fuse-utils' directory.</p>

  <p>One thing which isn't in the SourceForge tracking system (and is now very outdated):</p>

  <ul>
   <li>David Gardner has produced a patch to give <a href="fuse-xvideo.patch.bz2">XVideo support</a> for the Xlib UI, allowing arbitrary sized windows.</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section ("Related projects", "Are there any related projects?", <<<END_SECTION
  <ul>
   <li><a href="libspectrum.php">libspectrum</a> is the library used by Fuse to handle various file formats.</li>
   <li>Mike Wynne's ZX81 emulator, <a href="http://www.chuntey.com/eightyone/">EightyOne</a> is also using Fuse's Z80 core.</li>
   <li>Matthew Westcott's <a href="http://matt.west.co.tt/spectrum/jsspeccy/">JSSpeccy</a> uses a Z80 core based on translating Fuse's core to Javascript.</li>
   <li>Alexander Shabarshin's <a href="http://robots.ural.net/nedopc/sprinter/">SPRINT</a>, an emulator of the <a href="http://www.interface1.net/zx/clones/peters.html">Peters Plus</a> super-Speccy, the <a href="http://www.petersplus.com/sprinter/">Sprinter</a>. SPRINT is using Fuse's Z80 core for its CPU emulation.</li>
   <li><a href="http://sourceforge.net/projects/z80ex/">z80ex</a>, a Z80 emulation library based on Fuse's Z80 core, used by zemu and <a href="http://pocketspeccy.narod.ru/">PocketSpeccy</a>.</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_related ("libspectrum", "libspectrum.php");

fuse_footer ();
?>
